pandas/_libs/reduction.pyx

```diff
@@ -418,6 +415,13 @@ cdef class BlockSlider:
         self.dummy = frame[:0]
         self.index = self.dummy.index
 
+        # GH#35417 attributes we need to restore at each step in case
+        #  the function modified them.
+        mgr = self.dummy._mgr
+        self.orig_blklocs = mgr.blklocs
+        self.orig_blknos = mgr.blknos
```

Comment:
Is this fixing a specific bug? (but don't see any test added)

Reply:
without this we have other tests that fail in cython but not in python, but im not aware of any bugs this causes in master

Reply:
You mean that on master the `try: piece = f(chunk) ...` would fail, and thus be elevated to the python level (where it worked), but with this fix ensures those specific cases don't fail anymore an work on the cython level? 

Can you give an example of such a test?

Also, the issues number in the comment seems not directly related (well, it might be it was needed there as well, but since that's an open PR, that's a bit confusing reference)

Reply:
> Can you give an example of such a test?

```
FAILED pandas/tests/groupby/test_apply_mutate.py::test_mutate_groups - ValueError: shape mismatch: value array of shape (1,6) could not be broa...
```

Will attempt to clarify .